core/consensus/grandpa/vote_graph.hpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_CONSENSUS_GRANDPA_VOTE_GRAPH_HPP
+#define KAGOME_CORE_CONSENSUS_GRANDPA_VOTE_GRAPH_HPP
+
+#include "consensus/grandpa/chain.hpp"
+#include "consensus/grandpa/structs.hpp"
+
+namespace kagome::consensus::grandpa {
+
+  struct VoteGraph {
+    // graph entry
+    struct Entry {
+      BlockNumber number;
+      std::vector<BlockHash> ancestors;
+      std::vector<BlockHash> descendents;
+      // cumulative vote?
+    };
+
+    struct Subchain {
+      std::vector<BlockHash> hashes;
+      BlockNumber best_number;
+    };
+
+    virtual ~VoteGraph() = default;
+
+    using Condition = std::function<bool(/*vote weight*/)>;
+
+    explicit VoteGraph(const BlockInfo &base) : base_(base) {}
```

Comment:
Constructor will be moved to VoteGraphimpl

Reply:
Here I am indicating that I need to create VoteGraph given base info